client/reader/test/utils.js

```diff
@@ -8,30 +8,37 @@ import { showSelectedPost } from '../utils';
 jest.mock( 'page', () => ( {
 	show: jest.fn(),
 } ) );
-jest.mock( 'calypso/lib/redux-bridge', () => ( {
-	reduxGetState: function () {
-		return { reader: { posts: { items: {} } } };
-	},
-} ) );
 
 describe( 'reader utils', () => {
+	const dispatch = jest.fn();
+	const getState = () => ( {
+		reader: {
+			posts: {
+				items: {},
+			},
+		},
+	} );
+
 	beforeEach( () => {
 		page.show.mockReset();
 	} );
 
 	describe( '#showSelectedPost', () => {
 		test( 'does not do anything if postKey argument is missing', () => {
-			showSelectedPost( {} );
+			showSelectedPost( {} )( dispatch, getState );
 			expect( page.show ).not.toBeCalled();
 		} );
 
 		test( 'redirects if passed a post key', () => {
-			showSelectedPost( { postKey: { feedId: 1, postId: 5 } } );
+			showSelectedPost( { postKey: { feedId: 1, postId: 5 } } )( dispatch, getState );
```

Comment:
Interesting, so our mock data doesn't need to be populated with any fake posts?

Reply:
The actual post object is used only to check if the post is an x-post, nothing else. If the object is `undefined`, the x-post check returns `false`. We'd need mock post data only in a unit test that tests the x-post path specifically.